README.md

```diff
@@ -2,8 +2,35 @@
 
 A Rails engine for queuing and managing maintenance tasks.
 
+By ”maintenance task”, this project means a data migration, i.e. code that
+changes data in the database, often to support schema migrations. For example, in order to introduce a new `NOT NULL` column, it has to be added as nullable first, backfilled with values, before finally being changed to `NOT NULL`. This engine helps with the second part of this process, backfilling.
+
+Maintenance tasks are collection-based tasks, usually using Active Record, that update the data in your database. They can be paused or interrupted. If Active Record or Active Model is used, maintenance tasks will operate in batches to control the load on your database.
+
+Maintenance tasks aren't meant to happen on a regular basis. They're used as needed, or as one-offs. Normally maintenance tasks are ephemeral, so they are used briefly and then deleted.
+
+The Rails engine has a web-based UI for listing maintenance tasks, seeing their status, and starting, pausing and restarting them.
+
 [![Link to demo video](static/demo.png)](https://www.youtube.com/watch?v=BTuvTQxlFzs)
 
+## Should I Use Maintenance Tasks?
+
+Maintenance Tasks have a limited, specific job UI. While the engine can used to provide a user interface for other data changes, such as data changes for support requests for example, we recommend you use regular application code or [Rails administration engines][rails-admin-engines] for those use cases instead. These inevitably require more flexibility than this engine will be able to provide.
+
+If your task shouldn't run as an Active Job, it probably isn't a good match for Maintenance Tasks. If your task doesn't need to run in the background, consider a runner script instead. If your task doesn't need to be interruptible, consider a normal Active Job.
+
+Maintenance Tasks can be interrupted between iterations. If your task [isn't collection-based](#user-content-tasks-that-dont-need-a-collection) (no CSV file or database table) or has very large batches, it will get limited benefit from throttling (pausing between iterations) or interrupting. This might be fine, or the added complexity of Maintenance Tasks over normal Active Jobs may not be worthwhile.
+
+If your task updates schema instead of data, use a migration instead of a Maintenance Task.
+
+If your task happens regularly, consider Active Jobs with a scheduler or cron, [job-iteration jobs](https://github.com/shopify/job-iteration) and/or [custom rails_admin UIs][rails-admin-enginers] instead of a Maintenance Task. Maintenance Tasks should be ephemeral, to suit their intentionally limited UI.
+
+To create seed data for a new application, use Rails db/seeds.rb.
+
+If your application can't handle a half-completed migration, Maintenance Tasks are probably the wrong tool. Remember that they are intentionally pausable and can be cancelled halfway.
+
+[rails-admin-engines]: https://github.com/railsadminteam/rails_admin
```

Comment:
As discussed, let's link to the Ruby Toolbox here instead!
https://www.ruby-toolbox.com/categories/rails_admin_interfaces